File: app/src/main/java/com/example/expensetrcaker/ai/SpendingAI.java
package com.example.expensetrcaker.ai;

import androidx.compose.material.icons.Icons;
import androidx.compose.material.icons.rounded.Fastfood;
import androidx.compose.material.icons.rounded.Navigation;
import androidx.compose.material.icons.rounded.ShoppingBag;
import androidx.compose.material.icons.rounded.Star;
import androidx.compose.ui.graphics.Color;
import androidx.compose.ui.graphics.vector.ImageVector;
import com.example.expensetrcaker.data.TransactionEntity;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class SpendingAI {
    public static SpendingPersonality calculatePersonality(List<TransactionEntity> transactions) {
        if (transactions == null || transactions.isEmpty()) {
            return new SpendingPersonality(
                    "Fresh Start",
                    "You haven't recorded enough data yet.",
                    Icons.Rounded.getStar(Icons.Rounded.INSTANCE),
                    new Color(0xFF03DAC6L),
                    "Start adding transactions to unlock your spending personality!");
        }

        Map<String, Double> categoryTotals = new HashMap<>();
        for (TransactionEntity t : transactions) {
            if (t.isExpense()) {
                categoryTotals.put(t.getCategory(), categoryTotals.getOrDefault(t.getCategory(), 0.0) + t.getAmount());
            }
        }

        String topCategory = null;
        double maxAmount = -1;
        for (Map.Entry<String, Double> entry : categoryTotals.entrySet()) {
            if (entry.getValue() > maxAmount) {
                maxAmount = entry.getValue();
                topCategory = entry.getKey();
            }
        }

        if (topCategory == null) {
            return new SpendingPersonality(
                    "Balanced Pro",
                    "Your spending is well-distributed.",
                    Icons.Rounded.getStar(Icons.Rounded.INSTANCE),
                    new Color(0xFFBB86FCL),
                    "Great job! Your essentials and lifestyle spending are perfectly in sync.");
        }

        switch (topCategory) {
            case "Food":
                return new SpendingPersonality(
                        "The Gourmet",
                        "Your heart (and wallet) belongs to the kitchen.",
                        Icons.Rounded.getFastfood(Icons.Rounded.INSTANCE),
                        new Color(0xFFFF4081L),
                        "You spend 40% more on dining than average. Try meal prepping to save $200/mo.");
            case "Transport":
                return new SpendingPersonality(
                        "The Explorer",
                        "You're always on the move.",
                        Icons.Rounded.getNavigation(Icons.Rounded.INSTANCE),
                        new Color(0xFF00E5FFL),
                        "Commuting is your top expense. Consider a monthly pass to optimize your transit budget.");
            case "Shopping":
                return new SpendingPersonality(
                        "The Trendsetter",
                        "Style is your priority.",
                        Icons.Rounded.getShoppingBag(Icons.Rounded.INSTANCE),
                        new Color(0xFFFFC107L),
                        "Impulse buys are creeping up. Try the 24-hour rule before checking out!");
            default:
                return new SpendingPersonality(
                        "The Balanced Pro",
                        "Your spending is well-distributed.",
                        Icons.Rounded.getStar(Icons.Rounded.INSTANCE),
                        new Color(0xFFBB86FCL),
                        "Great job! Your essentials and lifestyle spending are perfectly in sync.");
        }
    }
}

File: app/src/main/java/com/example/expensetrcaker/ai/SpendingAI.kt
package com.example.expensetrcaker.ai

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Info
import androidx.compose.material.icons.filled.Savings
import androidx.compose.material.icons.filled.ShoppingCart
import androidx.compose.material.icons.filled.Star
import androidx.compose.material.icons.filled.TrendingUp
import androidx.compose.ui.graphics.Color
import com.example.expensetrcaker.data.TransactionEntity

object SpendingAI {
    @JvmStatic
    fun calculatePersonality(transactions: List<TransactionEntity>): SpendingPersonality {
        if (transactions.isEmpty()) {
            return SpendingPersonality(
                    name = "Newcomer",
                    description = "Start tracking to unlock insights",
                    insight = "Your financial journey begins with a single step.",
                    color = Color.Gray,
                    icon = Icons.Default.Info
            )
        }

        val expenses = transactions.filter { it.isExpense() }

        if (expenses.isEmpty()) {
            return SpendingPersonality(
                    name = "Saver",
                    description = "You are mindful of your spending.",
                    insight = "Great job! Keep building that emergency fund.",
                    color = Color(0xFF00E676),
                    icon = Icons.Default.Savings
            )
        }

        val categoryTotals =
                expenses.groupBy { it.getCategory() }.mapValues { entry ->
                    entry.value.sumOf { it.getAmount() }
                }

        val topCategory = categoryTotals.maxByOrNull { it.value }?.key ?: "General"

        return when (topCategory) {
            "Food" ->
                    SpendingPersonality(
                            name = "Foodie",
                            description = "You love good food!",
                            insight = "Consider cooking at home more often to save.",
                            color = Color(0xFFFF4081),
                            icon = Icons.Default.Star
                    )
            "Shopping" ->
                    SpendingPersonality(
                            name = "Shopaholic",
                            description = "Retail therapy is your go-to.",
                            insight = "Try the 24-hour rule before buying non-essentials.",
                            color = Color(0xFFFFC107),
                            icon = Icons.Default.ShoppingCart
                    )
            "Transport" ->
                    SpendingPersonality(
                            name = "Traveler",
                            description = "Always on the move.",
                            insight = "Look for commute passes or carpooling options.",
                            color = Color(0xFF00E5FF),
                            icon = Icons.Default.TrendingUp
                    )
            else ->
                    SpendingPersonality(
                            name = "Saver",
                            description = "You are mindful of your spending.",
                            insight = "Great job! Keep building that emergency fund.",
                            color = Color(0xFF00E676),
                            icon = Icons.Default.Savings
                    )
        }
    }
}
